cpp/ql/test/query-tests/Likely Bugs/Memory Management/UsingExpiredStackAddress/test.cpp

```diff
@@ -0,0 +1,199 @@
+struct S100 {
+  int i;
+  int* p;
+};
+
+static struct S100 s101;
+
+void escape1() {
+  int x;
+  s101.p = &x;
+}
+
+int simple_field_bad() {
+  escape1();
+  return *s101.p; // BAD
+}
+
+int simple_field_good() {
+  escape1();
+  return s101.i; // GOOD
+}
+
+int deref_p() {
+  return *s101.p;
+}
+
+int field_indirect_bad() {
+  escape1();
+  return deref_p(); // BAD [NOT DETECTED]
+}
+
+int deref_i() {
+  return s101.i;
+}
+
+int field_indirect_good() {
+  escape1();
+  return deref_i(); // GOOD
+}
+
+void store_argument(int *p) {
+  s101.p = p;
+}
+
+int store_argument_value() {
+  int x;
+  store_argument(&x);
+  return *s101.p; // GOOD
+}
+
+void store_address_of_argument(int x) {
+  s101.p = &x;
+}
+
+int store_argument_address() {
+  int x;
+  store_address_of_argument(x);
+  return *s101.p; // BAD
+}
+
+void address_escapes_through_pointer_arith() {
+  int x = 0;
```

Comment:
This ought really be an array, so that `p0 + 1` etc are valid addresses.

Reply:
Good point. Fixed in 033edc24f46c2eca7753891564c8557ececd0d84.